adapters/sinatra/lib/opentelemetry/adapters/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      module Middlewares
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            tracer.in_span(
+              env['PATH_INFO'],
+              attributes: { 'component' => 'http',
+                            'http.method' => env['REQUEST_METHOD'],
+                            'http.url' => env['PATH_INFO'] },
+              kind: :server,
+              with_parent_context: parent_context(env)
+            ) do |span|
+              app.call(env).tap { |resp| trace_response(span, env, resp) }
+            end
+          end
+
+          private
+
+          attr_reader :app
+
+          def parent_context(env)
+            OpenTelemetry.tracer_factory.http_text_format.extract(env)
+          end
+
+          def tracer
+            OpenTelemetry::Adapters::Sinatra::Adapter.tracer
+          end
+
+          def trace_response(span, env, resp)
+            status, _headers, _response_body = resp
+
```

Comment:
It similarly has the ability to opt-in to collecting response_headers: https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/sinatra/tracer_middleware.rb#L37-L39.

Should we preserve this behavior?